main.go

```diff
@@ -73,85 +73,94 @@ func main() {
+		submarinerInformers.WithNamespace(submSpec.Namespace))
+
 	start := func(context.Context) {
-		var ce cableengine.CableEngine
-		localCluster, err := util.GetLocalCluster(ss)
+		localCluster, err := util.GetLocalCluster(submSpec)
 		if err != nil {
-			klog.Fatalf("Fatal error occurred while retrieving local cluster %v", localCluster)
+			klog.Fatalf("Fatal error occurred while retrieving local cluster from %v: %v", submSpec, err)
 		}
-		localEndpoint, err := util.GetLocalEndpoint(ss.ClusterId, "ipsec", nil, ss.NatEnabled, append(ss.ServiceCidr, ss.ClusterCidr...))
+		localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterId, "ipsec", nil, submSpec.NatEnabled,
+			append(submSpec.ServiceCidr, submSpec.ClusterCidr...))
 		if err != nil {
-			klog.Fatalf("Fatal error occurred while retrieving local endpoint %v", localCluster)
+			klog.Fatalf("Fatal error occurred while retrieving local endpoint from %v: %v", submSpec, err)
 		}
-		ce = ipsec.NewEngine(append(ss.ClusterCidr, ss.ServiceCidr...), localCluster, localEndpoint)
 
-		tunnelController := tunnel.NewTunnelController(ss.Namespace, ce, kubeClient, submarinerClient,
-			submarinerInformerFactory.Submariner().V1().Endpoints())
+		var cableEngine cableengine.CableEngine = ipsec.NewEngine(append(submSpec.ClusterCidr, submSpec.ServiceCidr...), localCluster, localEndpoint)
 
-		var ds datastore.Datastore
+		tunnelController := tunnel.NewTunnelController(submSpec.Namespace, cableEngine, kubeClient, submarinerClient,
+			submarinerInformerFactory.Submariner().V1().Endpoints())
 
-		switch ss.Broker {
+		var datastore datastore.Datastore
+		switch submSpec.Broker {
 		case "phpapi":
-			secure, err := util.ParseSecure(ss.Token)
+			secure, err := util.ParseSecure(submSpec.Token)
 			if err != nil {
 				klog.Fatalf("Error parsing secure token: %v", err)
 			}
-			ds = phpapi.NewPHPAPI(secure.ApiKey)
+			datastore = phpapi.NewPHPAPI(secure.ApiKey)
 		case "k8s":
-			ds = subk8s.NewK8sDatastore(ss.ClusterId, stopCh)
+			datastore = subk8s.NewK8sDatastore(submSpec.ClusterId, stopCh)
 		default:
-			panic("No backend was specified")
+			klog.Fatalf("Invalid backend '%s' was specified", submSpec.Broker)
 		}
+
 		klog.V(6).Infof("Creating new datastore syncer")
-		dsSyncer := datastoresyncer.NewDatastoreSyncer(ss.ClusterId, ss.Namespace, kubeClient, submarinerClient, submarinerInformerFactory.Submariner().V1().Clusters(), submarinerInformerFactory.Submariner().V1().Endpoints(), ds, ss.ColorCodes, localCluster, localEndpoint)
+		dsSyncer := datastoresyncer.NewDatastoreSyncer(submSpec.ClusterId, submSpec.Namespace, kubeClient, submarinerClient,
+			submarinerInformerFactory.Submariner().V1().Clusters(), submarinerInformerFactory.Submariner().V1().Endpoints(), datastore,
+			submSpec.ColorCodes, localCluster, localEndpoint)
 
 		kubeInformerFactory.Start(stopCh)
 		submarinerInformerFactory.Start(stopCh)
 
 		klog.V(4).Infof("Starting controllers")
+
```

Comment:
nit: unrelated line change. I know it's a nit, but I try to avoid those, now, not very important, but in the future when we consider backports/reverts it can become messy.